src/feature/client/entrynodes.c

```diff
@@ -1771,7 +1773,7 @@ sample_reachable_filtered_entry_guards(guard_selection_t *gs,
            flags, smartlist_len(reachable_filtered_sample));
 
   if (smartlist_len(reachable_filtered_sample)) {
-    result = smartlist_choose(reachable_filtered_sample);
+    result = smartlist_get(reachable_filtered_sample, 0);
```

Comment:
Let's have a comment here that refers to proposal 310.

In particular, let's explain why this is safe, and why it won't just return the same guard over and over whether it's working or not.

Reply:
Ok, done. I've also noticed that functions calling first_reachable_filtered_entry_guard needed documentation updates. Function select_filtered_guard_for_circuit seemed said to look into confirmed guards while excluding them; I believe the documentation was wrong but the code was ok (i.e., we had to exclude confirmed guards).

Reply:
Hmm, I'd like to echo this. While prop310 is noted now, it's still not clear to me why the same guard wont be returned over and over. Perhaps some additional documentation or spec patch is in order here to increase comprehension?